conda_concourse_ci/concourse_config.py

```diff
@@ -458,14 +458,56 @@ def add_convert_task(self, subdir, docker_user=None, docker_pass=None):
                         'find . -name "converted-artifacts" -prune -o -path "*/{subdir}/*.tar.bz2" -print0 | xargs -0 -I file mv file converted-artifacts/{subdir}\n'  # NOQA
                         'find . -name "converted-artifacts" -prune -o -path "*/noarch/*.tar.bz2" -print0 | xargs -0 -I file mv file converted-artifacts/noarch\n'  # NOQA
-                'pushd converted-artifacts/{subdir} && cph t "*.tar.bz2" .conda && popd\n'
-                'pushd converted-artifacts/noarch && cph t "*.tar.bz2" .conda && popd\n'
+                        'pushd converted-artifacts/{subdir} && cph t "*.tar.bz2" .conda && popd\n'  # NOQA
+                        'pushd converted-artifacts/noarch && cph t "*.tar.bz2" .conda && popd\n'  # NOQA
                     .format(subdir=subdir)
                     ],
             }
         }
         self.plan.append({'task': 'convert .tar.bz2 to .conda', 'config': config})
 
+    def add_sign_artifacts_task(self, subdir, signing_key, docker_user=None, docker_pass=None):
+        _source = {
+            'repository': 'public.ecr.aws/y0o4y9o3/anaconda-pkg-build',
+            'tag': 'master-amd64',
+        }
+        if docker_user and docker_pass:
+            _source.update({
+                'username': docker_user,
+                'password': docker_pass
+            })
+        config = {
+            'platform': 'linux',
+            'inputs': [{'name': 'converted-artifacts'}],
+            'outputs': [{'name': 'signed-artifacts'}],
+            'image_resource': {
+                'type': 'docker-image',
+                'source': _source,
+            },
+            'run': {
+                'path': 'sh',
+                'args': [
+                    '-exc',
+                        'mkdir -p signed-artifacts/{subdir}\n'
+                        'mkdir -p signed-artifacts/noarch\n'
+                        'find . -name "signed-artifacts" -prune -o -path "*/{subdir}/*.tar.bz2" -print0 | xargs -0 -I file mv file signed-artifacts/{subdir}\n'  # NOQA
+                        'find . -name "signed-artifacts" -prune -o -path "*/noarch/*.tar.bz2" -print0 | xargs -0 -I file mv file signed-artifacts/noarch\n'  # NOQA
+                        'find . -name "signed-artifacts" -prune -o -path "*/{subdir}/*.conda" -print0 | xargs -0 -I file mv file signed-artifacts/{subdir}\n'  # NOQA
+                        'find . -name "signed-artifacts" -prune -o -path "*/noarch/*.conda" -print0 | xargs -0 -I file mv file signed-artifacts/noarch\n'  # NOQA
+                        'find . -name "signed-artifacts" -prune -o -path "*/{subdir}/*.sig" -print0 | xargs -0 -I file mv file signed-artifacts/{subdir}\n'  # NOQA
+                        'find . -name "signed-artifacts" -prune -o -path "*/noarch/*.sig" -print0 | xargs -0 -I file mv file signed-artifacts/noarch\n'  # NOQA
+                        'pushd signed-artifacts/{subdir} && TODO_SIGN_ARTIFACTS && popd\n'  # NOQA
```

Comment:
Any suggestions on how to implement the `TODO_SIGN_ARTIFACTS` piece?

I'm planning to pass in the ED25519 signing key secret as hex.  It's pretty simple (~10 lines of python) to use the `nacl` module to create a signing key with ascii hex secrets and write out a signature file - but I feel gross inlining python in shell this way.

Reply:
I also don't want to leak the signing secret in builder output, and I'm not sure if doing it this way would do that.

Reply:
[This](https://github.com/anaconda-distribution/rocket-platform/pull/694/files#diff-d54b5cc11713e8170ce86a4a318c1c10704505321aaf042ef968383035997a82R818-R832) is how I'm creating sig files in Prefect.

So, you can see I need the `json` module and `nacl`.  That should be it.